System.Containers/Layer.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Formats.Tar;
+using System.Security.Cryptography;
+
+namespace System.Containers;
+
+public record struct Layer
+{
+    public Descriptor Descriptor { get; private set; }
+
+    public string BackingFile { get; private set; }
+
+    public static Layer FromDirectory(string directory, string containerPath)
+    {
+        // Docker treats a COPY instruction that copies to a path like `/app` by
+        // including `app/` as a directory, with no leading slash. Emulate that here.
+        containerPath = containerPath.TrimStart(new char[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar });
+
+        DirectoryInfo di = new(directory);
+
+        string tempPath = Path.Join(Configuration.ArtifactRoot, "Temp");
+
+        Directory.CreateDirectory(tempPath);
+
+        long fileSize;
+        byte[] hash;
+
+        string tempTarballPath = Path.Join(tempPath, Path.GetRandomFileName());
+        using (FileStream fs = File.Create(tempTarballPath))
+        {
+            // using (GZipStream gz = new(fs, CompressionMode.Compress)) // TODO: https://github.com/dotnet/runtime/issues/65951#issuecomment-1145209082
```

Comment:
did this make it into preview 6? preemptive 🎆 in case it did

Reply:
Looks like yes per the tags on https://github.com/dotnet/runtime/commit/1d6f23a90767789b8c865b726a02fbffcf9d5346

However, I'm not going to do it now because it will break a thing in my e2e branch--there's a place where we're supposed to hash the uncompressed  tarball and some places where we're supposed to hash the compressed one, and right now they're conflated. Filed #29.